rust_src/crates/webrender/src/wrterm.rs

```diff
@@ -423,6 +424,15 @@ pub fn x_open_connection(
     _xrm_string: LispObject,
     _must_succeed: LispObject,
 ) -> LispObject {
+    let display = if unsafe { NILP(display) } {
+        let display = CString::new("").unwrap();
+        unsafe { build_string(display.as_ptr()) }
```

Comment:
`build_string("".as_ptr() as *const ::libc::c_char)` ?

Reply:
Thanks

Reply:
Oh, I think this is not safe. IIRC, the `String` is not nul-terminated while `CString` is.

Reply:
Hmm, I thought it's just a code style we're doing. So I just followed your suggestion.
Also I think I was using `CString` in the original code.

Reply:
Oh, we've used this in remacs several times so I thought it's ok.